packages/@react-aria/grid/src/useGridCell.ts

```diff
@@ -104,31 +126,47 @@ export function useGridCell<T, C extends GridCollection<T>>(props: GridCellProps
     isDisabled: state.collection.size === 0
   });
 
-  let onKeyDownCapture = (e: ReactKeyboardEvent) => {
-    if (!e.currentTarget.contains(e.target as Element) || state.isKeyboardNavigationDisabled) {
+  let onKeyDown = (e: ReactKeyboardEvent) => {
+    if (!e.currentTarget.contains(e.target as Element) || (!isEditing && state.isKeyboardNavigationDisabled)) {
       return;
     }
 
     let walker = getFocusableTreeWalker(ref.current);
     walker.currentNode = document.activeElement;
 
+    let isFocused = ref.current === document.activeElement;
+    let isNavigation = NAVIGATION_KEYS.has(e.key) || (e.altKey && e.key === 'Tab');
+
+    if (keyboardNavigationBehavior === 'tab' && isNavigation && !isFocused) {
```

Comment:
Known issue: When editing cells with a single focusable element, the arrow key events should be captured by the nested focusable element.

Reply:
Could we check if the nested focusable element (aka `document.activeElement`) is contained by the current ref as well? Noticed that arrow keys don't move the text cursor when focused on the cell text inputs when in the tab navigation table story.

Related to the above behavior, it feels odd IMO that the arrow keys in the arrow navigation table story move between the focusable elements in an editable cell instead of moving the text cursor of your focused textfield. I do see that the spec mentioned that this is the correct behavior... cc @majornista any opinions here?